src/URI.js

```diff
@@ -1615,6 +1616,13 @@
       segments.unshift('');
     }
 
+    // remove leading and trailing slashes
+    // from segment
+    for (var i=0; i < segments.length; i++) {
```

Comment:
the lines we're writing in are 1596, 1602, 1604, 1609. For lines 1602 and 1604 a single sanitization can be done on line 1607+. That makes three times `trimSlashes(v)` and one declaration of the function somewhere at the top (likely line 177, as the last internal utility function).

```js
function trimSlashes(text) {
  return text.replace(/^\/+|\/+$/g, '');
}
```
doesn't sound like making that if-from-hell any more complex… (looking at that code, a refactoring might be in order anyway…)